tests/components/modbus/test_modbus.py

```diff
@@ -17,56 +21,265 @@
+                    CONF_PORT: 5501,
+                    CONF_NAME: "modbusTest1",
+                    CONF_TIMEOUT: 30,
+                    CONF_DELAY: 10,
+                },
+                {
+                    CONF_TYPE: "serial",
+                    CONF_BAUDRATE: 9600,
+                    CONF_BYTESIZE: 8,
+                    CONF_METHOD: "rtu",
+                    CONF_PORT: "usb01",
+                    CONF_PARITY: "E",
+                    CONF_STOPBITS: 1,
+                    CONF_NAME: "modbusTest2",
+                },
+            ]
+        },
+    ],
+)
+async def test_multiple_config_modbus(hass, do_config):
+    """Run test for modbus."""
+
+    mock_sync = mock.MagicMock()
+    with mock.patch(
+        "homeassistant.components.modbus.modbus.ModbusTcpClient", return_value=mock_sync
+    ), mock.patch(
+        "homeassistant.components.modbus.modbus.ModbusSerialClient",
+        return_value=mock_sync,
+    ), mock.patch(
+        "homeassistant.components.modbus.modbus.ModbusUdpClient", return_value=mock_sync
+    ):
+        # mocking is needed to secure the pymodbus library is not called!
+        assert await async_setup_component(hass, DOMAIN, do_config)
+        await hass.async_block_till_done()
+        assert DOMAIN in hass.data
+
+        # the "if" should really be in the do_config above, but
+        # there are no easy way to have vol add default.
+        assert do_config[DOMAIN][0][CONF_NAME] in hass.data[DOMAIN]
+        assert do_config[DOMAIN][1][CONF_NAME] in hass.data[DOMAIN]
+
+
+@pytest.mark.parametrize(
+    "do_read_type",
+    [
+        "read_coils",
+        "read_discrete_inputs",
+        "read_input_registers",
+        "read_holding_registers",
     ],
 )
-async def test_config_modbus(hass, do_discovery, do_options, do_config):
+async def test_pymodbus_read(hass, do_read_type):
     """Run test for modbus."""
-    config = {
-        DOMAIN: {
-            **do_config,
-            **do_options,
-        }
+
+    # dummy config
+    config_hub = {
+        CONF_NAME: DEFAULT_HUB,
+        CONF_TYPE: "tcp",
+        CONF_HOST: "modbusTest",
+        CONF_PORT: 5001,
+        CONF_DELAY: 1,
+        CONF_TIMEOUT: 1,
     }
-    await base_config_test(
-        hass,
-        None,
-        "",
-        DOMAIN,
-        None,
-        None,
-        method_discovery=do_discovery,
-        config_modbus=config,
-    )
+
+    mock_sync = mock.MagicMock()
+    with mock.patch(
+        "homeassistant.components.modbus.modbus.ModbusTcpClient", return_value=mock_sync
+    ), mock.patch(
+        "homeassistant.components.modbus.modbus.ModbusSerialClient",
+        return_value=mock_sync,
+    ), mock.patch(
+        "homeassistant.components.modbus.modbus.ModbusUdpClient", return_value=mock_sync
+    ):
+        # mocking is needed to secure the pymodbus library is not called!
+        # simulate read_* functions in pymodbus library
+
+        hub = ModbusHub(config_hub)
+        print(hub)
```

Comment:
This doesn't seem complete. We don't want to use integration details directly in the tests. Set up the integration.